src/Task.cpp

```diff
@@ -1203,23 +1195,16 @@ void Task::addDependency (const std::string& uuid)
   if (uuid == get ("uuid"))
     throw std::string ("A task cannot be dependent on itself.");
 
-  // Store the dependency.
-  std::string depends = get ("depends");
-  if (depends != "")
-  {
-    // Check for extant dependency.
-    if (depends.find (uuid) == std::string::npos)
-      set ("depends", depends + ',' + uuid);
-    else
-    {
 #ifdef PRODUCT_TASKWARRIOR
-      Context::getContext ().footnote (format ("Task {1} already depends on task {2}.", get ("uuid"), uuid));
-#endif
-      return;
-    }
+  if (hasDependency (uuid))
+  {
+    Context::getContext ().footnote (format ("Task {1} already depends on task {2}.", get ("uuid"), uuid));
+    return;
   }
-  else
-    set ("depends", uuid);
+#endif
```

Comment:
To reflect the previous version exactly, I think only the footnote call should be within the controlled group of the `#ifdef` block. This could cause skipping over the return statement if `PRODUCT_TASKWARRIOR` macro is set to 0 (which is for current implementation of `taskd`).

Reply:
I had assumed that `PRODUCT_TASKWARRIOR` was un-set for timewarrior. Why is `taskd` using this at all?  And, does `taskd` print footnotes?

Reply:
The current implementation of the `taskd` daemon utilizes `Task` objects, for example while performing [on-server merging](https://github.com/GothenburgBitFactory/taskserver/blob/1.2.0/src/daemon.cpp) of updates.

Timewarrior and taskwarrior only share common base in the `libshared` repo.

I think these `#ifdef` controlled groups (I've seen them elsewhere as well) are used exactly to suppress the output on the `taskd` daemon which would stem from merging task updates triggering these code paths.

Reply:
Ah, I had the sense backward.  And I learned something about PRODUCT_TASKWARRIOR :)